app/helpers/favicons_helper.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module FaviconsHelper
```

Comment:
Overall, I like the idea here. I would maybe put this in `presenters/favicon_presenter.rb` and not even worry about the helper methods. I would also consider memoizing the results. If done in a class method, it means it would only be loaded into memory once. It would require you restarting the server if you add new icons, but would boost the performance of each page because it's not doing the extra file reads.

```ruby
class FaviconPresenter
  def self.favicons
    return @favicons if defined?(@favicons)
    pngs = Dir.glob("app/assets/images/favicons/*.png").map do |png|
      rel = png.include?("apple-touch-icon") ? "apple-touch-icon" : "shortcut icon"
      { path: "favicons/" + File.basename(png), rel: rel, type: "image/png" }
    end
    svgs = Dir.glob("app/assets/images/favicons/*.svg").map  do |svg|
      { path: "favicons/" + File.basename(svg), rel: 'shortcut icon', type: "image/svg" }
    end
    @favicons = pngs + svgs
  end
end

# in view
```haml
- FaviconPresenter.favicons.each do |favicon|
  = favicon_link_tag favicon[:path], rel: favicon[:rel], type: favicon[:type]
```